tonic/src/codec/compression.rs

```diff
@@ -2,20 +2,28 @@ use crate::{metadata::MetadataValue, Status};
 use bytes::{Buf, BytesMut};
 #[cfg(feature = "gzip")]
 use flate2::read::{GzDecoder, GzEncoder};
+use http::HeaderValue;
 use std::fmt;
 #[cfg(feature = "zstd")]
 use zstd::stream::read::{Decoder, Encoder};
 
 pub(crate) const ENCODING_HEADER: &str = "grpc-encoding";
 pub(crate) const ACCEPT_ENCODING_HEADER: &str = "grpc-accept-encoding";
 
+/// This should always match the cardinality of `CompressionEncoding`
+pub(crate) const COMPRESSION_ENCODINGS_LENGTH: usize = 2;
```

Comment:
To make this safer you could use `strum_macros::EnumCount` on `CompressionEncoding`. It adds a const so I think it should work in place of `COMPRESSION_ENCODINGS_LENGTH`. Not sure if you want to bring in strum just for this though, I'd understand if not since it's a fairly heavy dep.

If not, please add a big huge scary comment on the `CompressionEncoding` enum as well that says you must go update `COMPRESSION_ENCODINGS_LENGTH` if you add a new variant to the enum. Perhaps colocate the two things too.

Reply:
I like a macro too- I'll check strum- because otherwise this can cause very unpleasant unexpected behavior